spec/globus_client_spec.rb

```diff
@@ -119,12 +119,17 @@
       before do
         allow(described_class::Identity).to receive(:new).and_return(fake_identity)
         allow(GlobusClient::Authenticator).to receive(:token).and_return("a_token", "new_token")
-        allow(fake_identity).to receive(:exists?).once.and_raise(GlobusClient::UnexpectedResponse::UnauthorizedError)
-        allow(fake_identity).to receive(:exists?).once.and_return(true)
+        response_values = [:raise, true]
+        allow(fake_identity).to receive(:exists?).twice do
+          v = response_values.shift
+          (v == :raise) ? raise(GlobusClient::UnexpectedResponse::UnauthorizedError) : v
+        end
       end
 
-      it "retries Identity#exists?" do
+      it "fetches a new token and retries Identity#exists?" do
+        expect(client.config.token).to eq "a_token"
         expect(client.user_exists?(sunetid: "user")).to be true
+        expect(client.config.token).to eq "new_token"
```

Comment:
I like how you changed this test, and maybe we don't even care about the return value of the `user_exists?` call? If that's the case, you could try making a small tweak here to avoid the rubocop todo and reveal the intention of the test better:

```suggestion
        expect { client.user_exists?(sunetid: "user") }
          .to change(client.config, :token)
          .from("a_token")
          .to("new_token")
```

Though the rubocop ding isn't so bad. So keeping this approved either way.

Reply:
I like that.  Accepting and removing rubocop exception.